### PR TITLE
feat: Add example helm chart for non-secure and secure modes

### PR DESCRIPTION
**Pull request by lenny-intel**
Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A for deployment example**
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)

## Testing Instructions
**See README**

### CODE REVIEW

**Review by bnevis-i — DISMISSED** [review]
LGTM

**Review by AlexCuse — APPROVED** [review]

**Comment by bnevis-i** [comment]
Caveat: These helm charts assume a single worker node.  There should probably be some node placement constraints to enforce this, at least initially, so that support for distribution can be added on a service-by-service basis.

**Comment by lenny-intel** [comment]
> Do we have enough approvals to merge?

@bnevis-i,  Waiting on feedback from Matts from Intel.

**Comment by magerstam** [comment]
> > Do we have enough approvals to merge?
> 
> @bnevis-i, Waiting on feedback from Matts from Intel.

Sorry for the delay, reviewing now.

File: deployment/helm/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+## Uninstallation
+
+```bash
+helm uninstall edgex-jakarta -n edgex
+```
+
+## Test EdgeX
+
+EdgeX on kubernetes using NodePort type to expose services by default,You can use ping command to test whether the EdgeX services start successfully.
+
```
Comment (magerstam) [inline deployment/helm/README.md:32]: Typo: "default. You can..."

File: deployment/helm/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+For example, the edgex-core-data ping command format:
+
+```bash
+curl http://localhost:59880/api/v2/ping
+```
+
+
+## Access EdgeX UI
+
+With a modern browser, navigate to http://\<ExternalIP\>:30400.
```
Comment (magerstam) [inline deployment/helm/README.md:47]: Just to confirm, this is the static external NodePort that will be used for this?

File: deployment/helm/README.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+Use details see [EdgeX UI doc](https://github.com/edgexfoundry/edgex-ui-go)
+
+## Tips
+
+- This project is based on [docker-compose-no-secty.yml](https://github.com/edgexfoundry/edgex-compose/blob/jakarta/docker-compose-no-secty.yml),
+you can implement your customized version based on this.
+- Since the EdgeX pods communicates with each other through the kubernetes service name, make sure the kubernetes DNS is enabled.
+- Since other edgex services need to rely on consul to obtain configuration or register themselves to consul, other services cannot run normally until consul starts successfully.
```
Comment (magerstam) [inline deployment/helm/README.md:56]: Typo: EdgeX

File: deployment/helm/README.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+and a MetalLB load balancer (with an assigned IP of 192.168.122.200).
+Adaptations may be needed if your cluster uses something different.
+
+To install the Nginx ingress controller,
+follow the instructions at
+<https://kubernetes.github.io/ingress-nginx/deploy/#quick-start>
+
+To install MetalLB,
+follow the instructions at
+<https://metallb.universe.tf/installation/#installation-with-helm>
```
Comment (magerstam) [inline deployment/helm/README.md:108]: Do we have recommendations with the sample to use L2 or BGP mode for MetallB?

File: deployment/helm/README.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+```sh
+./tlsgen.sh
+./tlsinstall.sh
+```
+
+To install with security features enabled,
+set the `edgex.security.enabled` flag to `true` during installation.
+
+If necessary, uninstall the non-security one first.
+The helm chart is not coded to allow for dynamic switching in and out of secure mode.
```
Comment (magerstam) [inline deployment/helm/README.md:133]: so you need to do a helm uninstall and then re-install?

**Review by magerstam — APPROVED** [review]

File: deployment/helm/README.md (hunk 4 above)
Comment (bnevis-i) [reply deployment/helm/README.md:108]: No recommendation offered.  This is merely documentation as to what component was used to provide the IP.

File: deployment/helm/README.md (hunk 5 above)
Comment (bnevis-i) [reply deployment/helm/README.md:133]: Yes.  Uninstall then reinstall.  I haven't tested anything different.  I make no claims that his helm cart is production-worthy.  The regression test suite hasn't even been run against it.

File: deployment/helm/README.md (hunk 2 above)
Comment (bnevis-i) [reply deployment/helm/README.md:47]: Probably need to check all the port #'s.  All the custom nodeports were replaced with hostPort used by upstream EdgeX.  Primary motivation for this is that no validation has been done to make sure that EdgeX services run in a load-balanced configuration.  Once a service has been fully validated to run in HA mode we can put it back on a nodeport.

File: deployment/helm/README.md (hunk 1 above)
Comment (lenny-intel) [reply deployment/helm/README.md:32]: fixed

File: deployment/helm/README.md (hunk 3 above)
Comment (lenny-intel) [reply deployment/helm/README.md:56]: fixed

**Review by bnevis-i — APPROVED** [review]

**Pull request merged**